### PR TITLE
add filter wikidata by zoom

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/web/issues/497

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java [hunk 1]
```diff
@@ -81,16 +85,36 @@ public FeatureCollection getImagesById(long id, double lat, double lon) {
 		return new FeatureCollection(features.toArray(new Feature[0]));
 	}
 	
-	public FeatureCollection getWikidataData(String northWest, String southEast, String lang, Set<String> filters) {
+	public FeatureCollection getWikidataData(String northWest, String southEast, String lang, Set<String> filters, int zoom) {
 		String filterQuery = filters.isEmpty() ? "" : "AND poitype IN (" + filters.stream().map(s -> "'" + s + "'").collect(Collectors.joining(", ")) + ")";
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java:89]: Please avoid using **user-defined** `filters` inside string-formed SQL query. It is vulnerable for SQL injections. Using any strings as "'" + string "'" is an awful approach of forming SQL queries. Use SQL placeholders instead.

**Comment by vshcherb** [comment]
Need to test wikidata coords filter first

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java [hunk 2]
```diff
@@ -45,6 +45,10 @@ public class WikiService {
 
 	private static final int LIMIT_QUERY = 1000;
 	private static final int LIMITI_QUERY = 25;
+	
+	private static final Map<Integer, Set<String>> EXCLUDED_POI_TYPES_BY_ZOOM = Map.of(14, Set.of("shop"));
+	private static final Map<Integer, Set<String>> EXCLUDED_POI_SUBTYPES_BY_ZOOM = Map.of(14, Set.of("artwork", "commercial", "memorial", "monument"));
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java:50]: Some artwork / memorial could be very popular https://www.wikidata.org/wiki/Q185382

**Pull request merged**